packages/php-wasm/node/src/test/php-dynamic-loading.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { loadNodeRuntime } from '..';
+import { PHP, SupportedPHPVersions } from '@php-wasm/universal';
+import path from 'path';
+import fs from 'fs';
+import { createServer } from 'net';
+
+vi.mock('path', async (originalPath) => ({
+	...(await originalPath()),
+	dirname: vi.fn((input) => path.resolve(input, '../../../../')),
+}));
+
+describe.each(SupportedPHPVersions)('PHP %s', (phpVersion) => {
+	describe('XDebug', () => {
+		let php: PHP;
+		beforeEach(async () => {
+			php = new PHP(
+				await loadNodeRuntime(phpVersion as any, { withXdebug: true })
+			);
+		});
+
+		it('does not load dynamically by default', async () => {
+			php = new PHP(await loadNodeRuntime(phpVersion as any));
+
+			const result = await php.runStream({
+				code: `<?php
+                    var_dump(extension_loaded('xdebug'));`,
+			});
+
+			expect(await result.stdoutText).toEqual('bool(false)\n');
+		});
+
+		it('supports dynamic loading', async () => {
```

Comment:
let's also test `var_dump()` output – it should be fancier when xdebug is enabled